docs/core/deploying/native-aot/diagnostics.md

```diff
@@ -1,70 +1,78 @@
+The .NET CLI tooling (`dotnet` SDK) and Visual studio offer separate commands for `build` and
+`publish`. `build` (or `Start` in Visual Studio) will use CoreCLR. Only `publish` will create a
+Native AOT application.  Publishing your app as native AOT produces an app that has been
+ahead-of-time (AOT) compiled to native code. As mentioned above, this means that not all diagnostic
+tools will work seamlessly with published native AOT applications in .NET 8. However, all .NET
+diagnostic tools are available for developers during the application building stage. We recommend
+developing, debugging, and testing the applications as usual and publish the working app with native
+AOT as one of the last steps.
 
-### Production debugging
+## Debugging
 
-Typical production debugging scenarios are done through logging and tracing and are [supported](#observability) in native AOT. Low-level debugging, using platform debuggers like WinDbg or Visual Studio on Windows, and gdb or lldb on Unix-like systems, can be used in native AOT. For this case, it's critical that the corresponding symbol file for the application is available.
+When running your app during development, like inside Visual Studio, or with `dotnet run`, `dotnet build`, or `dotnet test`, applications will run on CoreCLR by default. However, as long as `PublishAot` is present in the project file the behavior should be the same between CoreCLR and Native AOT. This allows you to use the standard Visual Studio managed debugging engine for development and testing.
 
-The native AOT compiler generates information about line numbers, types, locals, and parameters. The native debugger allows inspecting stack trace and variables, stepping into or over source lines, or setting line breakpoints.
+After publishing, Native AOT applications are true native binaries. The managed debugger will not work on them. However, the Native AOT compiler generates fully native executable files that can be debugged by native debuggers on your platform of choice (e.g. WinDbg or Visual Studio on Windows, and gdb or lldb on Unix-like systems).
+
+The NativeAOT compiler generates information about line numbers, types, locals and parameters. The native debugger will let you inspect stack trace and variables, step into/over source lines, or set line breakpoints.
+
+To debug managed exceptions, set a breakpoint on the `RhThrowEx` method -- this method is called whenever a managed exception is thrown.
 
 Collecting a [dump](../../diagnostics/dumps.md) file for a native AOT application involves some manual steps in .NET 8.
 
-### Heap analysis
+### Visual Studio-specific notes
+
+You can launch a NativeAOT-compiled executable under the VS debugger by opening it in the Visual Studio IDE. In the `File` menu, choose `Open Project/Solution...` and navigate to the native executable. You can set breakpoints as needed. To start debugging the EXE, choose the `Start Debugging` option from the Debug menu.
```

Comment:
Link to https://learn.microsoft.com/visualstudio/debugger/how-to-debug-an-executable-not-part-of-a-visual-studio-solution?